sourcegraph/sourcegraph
# search ui: lucky search toggle menu

Shows toggle menu for lucky search. Currently, this will only be shown if `defaultPatternType` is `lucky`. Enabling lucky search via this menu will hide the other toggles.

![image](https://user-images.githubusercontent.com/206864/188702800-3ee7dfbc-eef8-49dc-8413-3a9f90d56be1.png)

![image](https://user-images.githubusercontent.com/206864/188702855-4fb74092-c8db-4777-bf06-071c93434c24.png)

## Test plan

- Verify the icon only shows if `defaultPatternType` is `lucky`
- Verify clicking icon opens menu
- Verify clicking on the menu options turns lucky search on and off
- Verify other toggles are only shown if lucky search is off

## App preview:

- [Web](https://sg-web-jp-luckysearchtogglemenu.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-udzkfnujgl.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
